TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/Turret.java

```diff
@@ -25,6 +25,48 @@ public Turret(AnalogInput left_potentiometer, AnalogInput right_potentiometer, S
         this.shooter = shooter;
     }
 
+    public void setGrabber(int mode){
+        if (mode == 0){
+            left_lift.setPower(0);
+            right_lift.setPower(0);
+        } else if (mode == 1){
+            left_lift.setPower(-0.2);
+            right_lift.setPower(-0.2);
+        } else if (mode == 2){
+            left_lift.setPower(0.2);
+            right_lift.setPower(0.2);
+        }
+    }
+
+    public void setFinger(int pos){
+        // TODO Find init, hold, extended positions of finger
+        if (pos == 0){
+            finger.setPosition(0);
+        } else if (pos == 1){
+            finger.setPosition(0.8);
+        } else if (pos == 2){
+            finger.setPosition(1);
+        }
+    }
+
+    public void setLift(int pos){
+        // TODO Determine Proportional Gain
+        double proportional = 0.6;
+        double[] voltage_positions = new double[] {0, 0.669, 0.776};
+        double error = getPotenPos()[0] - voltage_positions[pos];
+        while (error != 0){
```

Comment:
```java
        while (error != 0){
```
is not a good idea: First off, you block up the entire main loop to do this operation. Secondly, you wait until the error is *exactly* zero, which is going to be nearly impossible in reality.

Reply:
I'm going to add a "deadband" on either side of 0 so that it's actual possible for it to stop. Also, I should probably use the scheduler for this shouldn't I?

Reply:
Yes this should be done through the scheduler. It should also have its own event class so other code can respond when it's completed.